src/services/llm/utils.py

```
@@ -82,12 +84,38 @@ def is_local_llm_server(base_url: str) -> bool:
-    # Check for local hostname indicators
-    for host in LOCAL_HOSTS:
-        if host in base_url_lower:
+    # Extract hostname/IP from URL
+    try:
+        from urllib.parse import urlparse
+        parsed = urlparse(base_url)
+        hostname = parsed.hostname or parsed.netloc
+        if not hostname:
+            return False
+        hostname_lower = hostname.lower()
+    except Exception:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)